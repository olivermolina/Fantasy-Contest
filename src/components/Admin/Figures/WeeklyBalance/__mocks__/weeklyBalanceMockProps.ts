export const weeklyBalanceMockData = [
  {
    agent: null,
    id: '1',
    username: 'jdoe',
    firstname: 'John',
    lastname: 'Doe',
    referral: '',
    isFirstDeposit: false,
    mondayBalance: 20,
    tuesdayBalance: 0,
    wednesdayBalance: -35,
    thursdayBalance: 0,
    fridayBalance: 0,
    saturdayBalance: 0,
    sundayBalance: 20,
    deposits: 50,
    withdrawals: 6,
    totalWeekBalance: 5,
    pendingTotal: 20,
    cashAmount: 0,
    withdrawable: 0,
    creditAmount: 0,
    totalBalance: 0,
    isActive: true,
  },
  {
    agent: null,
    id: '2',
    username: 'jwick',
    firstname: 'John',
    lastname: 'Wick',
    referral: '',
    isFirstDeposit: false,
    mondayBalance: 20,
    tuesdayBalance: 0,
    wednesdayBalance: -35,
    thursdayBalance: 0,
    fridayBalance: 0,
    saturdayBalance: 0,
    sundayBalance: 20,
    deposits: 50,
    withdrawals: 6,
    totalWeekBalance: 5,
    pendingTotal: 20,
    cashAmount: 0,
    withdrawable: 0,
    creditAmount: 0,
    totalBalance: 0,
    isActive: true,
  },
];
