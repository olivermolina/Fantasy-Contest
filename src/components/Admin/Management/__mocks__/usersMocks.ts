import { faker } from '@faker-js/faker';
import { UserType } from '@prisma/client';

export const usersMock = [
  {
    id: '1',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [
      {
        id: '1',
        userId: '1',
        code: 'cod1',
      },
      {
        id: '2',
        userId: '1',
        code: 'code2',
      },
    ],
    type: UserType.AGENT,
  },
  {
    id: '2',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [],
    type: UserType.PLAYER,
  },
  {
    id: '3',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [],
    type: UserType.PLAYER,
  },
  {
    id: '4',
    email: faker.internet.email(),
    username: faker.internet.userName(),
    referral: faker.internet.userName(),
    referralCodes: [],
    type: UserType.PLAYER,
  },
];
