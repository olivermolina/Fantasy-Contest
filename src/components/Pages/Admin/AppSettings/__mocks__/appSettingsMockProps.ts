export const appSettingsMockProps = {
  appSettings: {
    BONUS_CREDIT_FREE_ENTRY_EQUIVALENT: '25',
    STAKE_TYPE_FREE_ENTRY: 'ALL_IN,INSURED',
    NUMBER_OF_PLAYERS_FREE_ENTRY: '2,3,4',
    MAX_MATCH_DEPOSIT_AMOUNT: '100',
    DEPOSIT_AMOUNT_OPTIONS: '10,25,50,75,100,200',
    RELOAD_BONUS_TYPE: 'FLAT',
    RELOAD_BONUS_AMOUNT: '0',
    MAX_RETENTION_BONUS: '50',
    RETENTION_BONUS_WEEKLY_CHANCE: '1',
    RETENTION_BONUS_MATCH_MULTIPLIER: '1',
    REFERRAL_CREDIT_AMOUNT: '25',
    REFERRAL_CUSTOM_TEXT:
      'Refer a friend from the link or code in your profile and get $25 in bonus credit!!',
    MIN_BET_AMOUNT: '1',
    MAX_BET_AMOUNT: '50',
    MAX_MARKET_ODDS: '150',
    MIN_MARKET_ODDS: '-150',
    initial: false,
    CHALLENGE_PROMO_MESSAGE:
      'Pick 2-4 players. Predict if they will get MORE or LESS than their projection.',
    REPEAT_ENTRIES_LIMIT: '1',
  },
  onSubmit: () => console.log('onSubmit'),
};
