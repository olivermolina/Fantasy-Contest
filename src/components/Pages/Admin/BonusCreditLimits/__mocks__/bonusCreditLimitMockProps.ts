import { BetStakeType } from '@prisma/client';

export const bonusCreditLimitMockProps = {
  onSubmit: () => console.log('submit'),
  data: {
    signupFreeEntry: false,
    bonusCreditFreeEntryEquivalent: 50,
    bonusCreditLimits: [
      {
        contestCategoryId: '1',
        numberOfPicks: 2,
        id: '30387acd-660c-46f0-a9e6-af87c007e6fb',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 25,
      },
      {
        contestCategoryId: '2',
        numberOfPicks: 3,
        id: '316555e5-b447-468a-b57f-381bbf64919a',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 50,
      },
      {
        contestCategoryId: '3',
        numberOfPicks: 4,
        id: 'fc063fcd-1a25-46e2-9361-d947eec1f9a9',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 40,
      },
      {
        contestCategoryId: '4',
        numberOfPicks: 5,
        id: 'a2542740-e90f-4763-8a3a-f7afba68898c',
        enabled: true,
        stakeTypeOptions: [BetStakeType.ALL_IN, BetStakeType.INSURED],
        bonusCreditFreeEntryEquivalent: 20,
      },
    ],
  },
  openAddUsersFreeEntryConfirmDialog: () => console.log('open'),
};
