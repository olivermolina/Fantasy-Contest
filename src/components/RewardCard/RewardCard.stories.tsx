import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import Reward from '.';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Lockspread/RewardCard',
  component: Reward,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
    isFirstRewardCard: {
      options: [true, false],
      control: { type: 'radio' },
    },
  },
} as ComponentMeta<typeof Reward>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Reward> = (args) => <Reward {...args} />;

export const RewardCard = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
RewardCard.args = {
  banner: {
    id: 'test',
    text: 'LockSpread will match your first deposit up to $200!',
    priority: 1,
    created_at: new Date(),
    updated_at: new Date(),
  },
};
