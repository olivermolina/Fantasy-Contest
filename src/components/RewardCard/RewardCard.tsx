import * as React from 'react';
import { AppSettings, Banner } from '@prisma/client';

interface Props {
  banner: Banner & { appSetting: AppSettings | null };
  handleClick: () => void;
}

export default function RewardCard(props: Props) {
  const { banner, handleClick } = props;
  return (
    <div
      className={
        'flex flex-row gap-4 justify-start items-center bg-[#1a395b] p-4 rounded-lg cursor-pointer'
      }
      onClick={handleClick}
    >
      <img
        className="object-cover w-28 h-28"
        src={'/assets/images/ico_gift_w_block.svg'}
        alt="Gift"
      />
      <div className={'flex flex-col gap-2 max-w-[300px]'}>
        <span className={'text-white text-3xl font-bold'}>
          ${banner.appSetting?.value}
        </span>
        <span className={'text-justify text-[#69A4F3]'}>{banner.text}</span>
      </div>
    </div>
  );
}
