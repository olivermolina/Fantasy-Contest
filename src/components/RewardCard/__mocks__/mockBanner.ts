import { AppSettingName } from '@prisma/client';

export const mockBanner = {
  id: 'test',
  text: 'LockSpread will match your first deposit up to $200!',
  priority: 1,
  created_at: new Date(),
  updated_at: new Date(),
  appSettingId: 'test',
  appSetting: {
    id: 'test',
    name: AppSettingName.MAX_MATCH_DEPOSIT_AMOUNT,
    value: '50',
  },
};
