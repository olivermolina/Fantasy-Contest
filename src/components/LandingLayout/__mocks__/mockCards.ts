import { contestCategoriesMock } from '~/components/Pages/Admin/ManageFreeSquarePromotion/__mocks__/contestCategoriesMock';

export const mockCards = [
  {
    id: '1',
    onClickLess: () => alert('clicked'),
    onClickMore: () => alert('clicked'),
    stat: 'Passing Yards',
    value: 99.5,
    gameInfo: 'LA@DEN',
    playerName: 'Patrick Mahomes',
    image: '/assets/images/dallas-goedert.png',
    playerPosition: 'QB',
    playerTeam: 'KC',
    freeSquare: {
      id: '1',
      discount: 98,
      pickCategories: contestCategoriesMock,
      maxStake: 25,
      freeEntryEnabled: false,
    },
  },
  {
    id: '2',
    onClickLess: () => alert('clicked'),
    onClickMore: () => alert('clicked'),
    stat: 'Passing Yards',
    value: 99.5,
    gameInfo: 'LA@DEN',
    playerName: 'Patrick Mahomes',
    image: '/assets/images/patrick-mahomes.png',
    playerPosition: 'QB',
    playerTeam: 'KC',
    freeSquare: null,
  },
  {
    id: '3',
    onClickLess: () => alert('clicked'),
    onClickMore: () => alert('clicked'),
    stat: 'Passing Yards',
    value: 99.5,
    gameInfo: 'LA@DEN',
    playerName: 'Patrick Mahomes',
    image: '/assets/images/tyrod-taylor.png',
    playerPosition: 'QB',
    playerTeam: 'KC',
    freeSquare: null,
  },
  {
    id: '4',
    onClickLess: () => alert('clicked'),
    onClickMore: () => alert('clicked'),
    stat: 'Passing Yards',
    value: 99.5,
    gameInfo: 'LA@DEN',
    playerName: 'Patrick Mahomes',
    image: '/assets/images/tyrod-taylor.png',
    playerPosition: 'QB',
    playerTeam: 'KC',
    freeSquare: null,
  },
];
