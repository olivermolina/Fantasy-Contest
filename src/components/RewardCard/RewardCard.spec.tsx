import React from 'react';
import { render } from '@testing-library/react';
import RewardCard from './RewardCard';
import { mockBanner } from './__mocks__/mockBanner';

describe('RewardCard', () => {
  it('should render', () => {
    const props = {
      banner: mockBanner,
      handleClick: jest.fn(),
    };
    const { container } = render(<RewardCard {...props} />);
    expect(container).toMatchSnapshot();
  });
});
