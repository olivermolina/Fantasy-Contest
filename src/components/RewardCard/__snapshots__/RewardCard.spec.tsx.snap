// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewardCard should render 1`] = `
<div>
  <div
    class="flex flex-row gap-4 justify-start items-center bg-[#1a395b] p-4 rounded-lg cursor-pointer"
  >
    <img
      alt="Gift"
      class="object-cover w-28 h-28"
      src="/assets/images/ico_gift_w_block.svg"
    />
    <div
      class="flex flex-col gap-2 max-w-[300px]"
    >
      <span
        class="text-white text-3xl font-bold"
      >
        $
        50
      </span>
      <span
        class="text-justify text-[#69A4F3]"
      >
        LockSpread will match your first deposit up to $200!
      </span>
    </div>
  </div>
</div>
`;
